federation/src/auth/JWE.ts

```diff
@@ -0,0 +1,103 @@
+import { createPrivateKey, createPublicKey } from 'crypto'
+
+import { compactDecrypt, compactVerify, CompactSign, CompactEncrypt } from 'jose'
+
+import {
+  crypto_sign_ed25519_pk_to_curve25519,
+  crypto_sign_ed25519_sk_to_curve25519,
+} from 'sodium-native'
+
+const crypto_scalarmult_curve25519_BYTES = 32
+
+export const verifyToken = async (
+  token: string,
+  keyPair: { publicKey: Buffer; privateKey: Buffer },
+  nonce: number | null = null,
+) => {
+  const pubKeyX = Buffer.alloc(crypto_scalarmult_curve25519_BYTES)
+  const privKeyX = Buffer.alloc(crypto_scalarmult_curve25519_BYTES)
+  crypto_sign_ed25519_pk_to_curve25519(pubKeyX, keyPair.publicKey.subarray(0, 32))
```

Comment:
Please use the constant definition from the used library instead of a hardcoded value of 32